projects/lib/src/openingsuite.cpp

```diff
@@ -129,37 +136,48 @@ bool OpeningSuite::initialize()
 			if (pos.pos == -1)
 				break;
 
-			int i = Mersenne::random() % (m_filePositions.size() + 1);
-			if (i == m_filePositions.size())
-				m_filePositions.append(pos);
-			else
-			{
-				m_filePositions.append(m_filePositions.at(i));
-				m_filePositions[i] = pos;
-			}
+			m_filePositions.append(pos);
+		}
+
+		// use a Knuth shuffle to generate a random permutation
+                for (int i = 0; i <= m_filePositions.size() - 2; i++)
+		{
+			int j = i + Mersenne::random() % (m_filePositions.size() - i);
+			std::swap(m_filePositions[i], m_filePositions[j]);
 		}
```

Comment:
why not something like
```
auto rng = [](){return Mersenne::random();};
std::shuffle(m_filePositions.begin(), m_filePositions.end(), rng);
```

Reply:
The honest answer is that I wasn't aware. Yet, I don't think this is suitable for us as cppreference mentions that: ```Note that the implementation is not dictated by the standard, so even if you use exactly the same RandomFunc or URBG you may get different results with different standard library implementations. ``` while we would like the same random order on cutechess-cli being compiled in different environments.